### PR TITLE
Security and build improvements.

### PR DESCRIPTION
**Pull request by z-glitch**
* Bump php version from 7.4 (unmaintained) to 8.2
* Fix php8 warning about non-default params following default params
* Upgrade installed packages during build
* Tell Docker to ignore builds folder
* Fix template formatting bug

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -13,6 +13,7 @@ RUN a2enmod rewrite
 
 # install the PHP extensions we need
 RUN apt-get update \
+        && apt-get -y --no-install-recommends dist-upgrade \
```
Comment (julianxhokaxhiu) [inline Dockerfile]: May I ask why did you add this line?

File: src/Helpers/Build.php [hunk 2]
```diff
@@ -285,7 +285,7 @@ protected function _getChannel( $token, $type, $version ) {
          * @param string $fileName The name of the file
          * @return string The absolute URL for the file to be downloaded
          */
-        protected function _getUrl( $fileName = '', $basePath ) {
+        protected function _getUrl( $fileName, $basePath ) {
```
Comment (julianxhokaxhiu) [inline src/Helpers/Build.php:288]: Will `$fileName` be empty by default?

File: Dockerfile (hunk 1 above)
Comment (z-glitch) [reply Dockerfile]: I added this to ensure other system packages are up-to-date, since many base images have a slower release cadence that may leave unapplied security updates.

File: src/Helpers/Build.php (hunk 2 above)
Comment (z-glitch) [reply src/Helpers/Build.php:288]: All references to _getUrl already specified $fileName, so assigning it a default value was unnecessary at this time.  I made the change because it is a syntax error in PHP8 to follow a parameter having a default value with one that lacks a default value, and this seemed the least invasive fix.

File: Dockerfile (hunk 1 above)
Comment (julianxhokaxhiu) [reply Dockerfile]: Better keep the docker image as it comes by default from the Registry. Can you please remove this line?

File: src/Helpers/Build.php (hunk 2 above)
Comment (julianxhokaxhiu) [reply src/Helpers/Build.php:288]: Understood, did you manage to test this code already?

File: Dockerfile (hunk 1 above)
Comment (z-glitch) [reply Dockerfile]: Removed at maintainer's request.

File: src/Helpers/Build.php (hunk 2 above)
Comment (z-glitch) [reply src/Helpers/Build.php:288]: Yes, I tested it with some personal android builds before submitting the PR.

**Review by julianxhokaxhiu — APPROVED** [review]

**Comment by julianxhokaxhiu** [comment]
Thanks a lot :) Really appreciated!
